cmd/faucet/faucet.go

```diff
@@ -440,6 +446,13 @@ func (f *faucet) startStack(genesis *genesisT.Genesis, port int, enodes []*discv
 	// Assemble the Ethereum light client protocol
 	cfg := eth.DefaultConfig
 	cfg.SyncMode = downloader.LightSync
+	if *syncmodeFlag == "fast" {
+		cfg.SyncMode = downloader.FastSync
+		cfg.ProtocolVersions = eth.DefaultProtocolVersions
+	} else if *syncmodeFlag == "full" {
+		cfg.SyncMode = downloader.FullSync
+		cfg.ProtocolVersions = eth.DefaultProtocolVersions
+	}
```

Comment:
We might want to be explicit here that we don't handle anything else than "light".

Maybe we can add something like:
```suggest
} else if *syncmodeFlag != "light" {
  return fmt.Error("Supported Sync modes are one of [light | fast | full].")
}
```

Reply:
Yea, good call. See https://github.com/etclabscore/core-geth/pull/268/commits/c883a4a221a589ea62f3d2d14669e7435885a5d3, where I've enforced this in `auditFlagUse` and refactored a little to both improve legibility and remove useless `backend` construction (which is only used for `ethstats` reporting).

Reply:
Nice refactor man. It's more readable now